.github/workflows/build-all.yml

```diff
@@ -219,6 +221,21 @@ jobs:
 
         cat *.sha256 | sha256sum --check --status
 
+    - name: Install Cosign
+      if: ${{ github.ref_type == 'tag' }}
+      uses: sigstore/cosign-installer@v3.1.1
+
+    - name: Sign Release
+      if: ${{ github.ref_type == 'tag' }}
+      run: |
+        cosign version
+        cosign sign-blob --yes --bundle "release/fossa_${{ steps.get-version.outputs.VERSION }}_linux_amd64.bundle" "release/fossa_${{ steps.get-version.outputs.VERSION }}_linux_amd64.tar.gz"
+
+    - name: Verify Signature
+      if: ${{ github.ref_type == 'tag' }}
+      run:
+        cosign verify-blob --bundle "release/fossa_${{ steps.get-version.outputs.VERSION }}_linux_amd64.bundle" --certificate-oidc-issuer "https://token.actions.githubusercontent.com" --certificate-identity "https://github.com/$GITHUB_WORKFLOW_REF" "release/fossa_${{ steps.get-version.outputs.VERSION }}_linux_amd64.tar.gz"
```

Comment:
I chose this because we would get multiple bundle files, one for each binary. I figured it would be more straightforward to for a customer verify the entire tarball rather than each individual binary. Cosign doesn't actually stick any new info in the binary itself like Mac OS does.

Reply:
On further reflection it make it easier to verify a binary that gets separated from the tarball. So I could see changing this if you feel that I should.

Reply:
I think the unit users really care about is the binary, so yes I would recommend we sign the binaries instead. In particular this allows users to run more flexible workloads: for example they can cache `fossa` between CI runs/steps & validate it after retrieving from the cache, validate the version of `fossa` installed with the shell installer, etc.

Reply:
I've altered this to produce a bundle for each individual binary. If you want to check the results, this build has them: https://github.com/fossas/fossa-cli/actions/runs/5741065721

To verify you can should be able to use this command:
```
cosign verify-blob --bundle fossa.bundle --certificate-oidc-issuer='https://token.actions.githubusercontent.com' --certificate-identity='https://github.com/fossas/fossa-cli/.github/workflows/build-all.yml@refs/tags/test-tag-delete-me' ./fossa
```